src/popup/pages/NetworkForm.vue

```diff
@@ -149,22 +151,34 @@ export default defineComponent({
     }
 
     async function addOrUpdateNetwork() {
-      if (hasErrors.value) {
-        return;
+      if (await $validator.validateAll()) {
+        root.$store.commit('setUserNetwork', {
+          ...newNetwork.value,
+          index: newNetwork.value.index,
+        });
+        await selectNetwork(newNetwork.value.name);
+        goBack();
       }
-
-      root.$store.commit('setUserNetwork', {
-        ...newNetwork.value,
-        index: newNetwork.value.index,
-      });
-      await selectNetwork(newNetwork.value.name);
-      goBack();
     }
 
-    onMounted(() => {
-      const { name } = root.$route.params;
-      if (name) {
-        newNetwork.value = { ...networks.value[name] };
+    onMounted(async () => {
+      const { params, query } = root.$route;
+
+      if (isEdit) {
+        newNetwork.value = { ...networks.value[params.name] };
+      } else if (Object.keys(query).length) {
+        // Fields that values are allowed to be passed from the URL query to the form model
+        const keys: (keyof INetworkBase)[] = ['name', 'url', 'middlewareUrl', 'compilerUrl'];
+        keys.forEach((key) => {
+          const val = query[key];
+
+          if (val && typeof val === 'string') {
+            set(newNetwork.value, key, val);
+          }
+        });
+
+        await nextTick();
+        $validator.validateAll();
```

Comment:
It seems like validation is not working correctly, using the link below I can create a network whose name is bigger than 15 characters

https://feat-create-new-network-from-a-link.wallet.z52da5wt.xyz/more/settings/networks/add?name=ysssdass12s3dasdo&url=https%3A%2F%2Ftest.com

Reply:
Nice finding. I added custom veeValidate rule for this.